### PR TITLE
Fix #21679: Added `metadata ingest-dbt` CLI Command for Direct DBT Artifacts Ingestion

### LINKED ISSUE #21679: Create CLI package for dbt project
**Affected module**
Ingestion Framework

**Describe the bug**
Create a tooling which users can run on their dbt project which will push metadata of dbt to OpenMetadata

### PR DESCRIPTION
**Pull request by Khairajani**
### Description:

This PR introduces a new CLI command **`metadata ingest-dbt`** that enables direct ingestion of dbt artifacts to OpenMetadata without requiring separate workflow configuration files. This streamlines the dbt metadata ingestion process by reading configuration directly from the dbt project's `dbt_project.yml` file.

#### 📋 **Usage:**
```bash
# If in DBT project directory:
metadata ingest-dbt

# If in a different directory:
metadata ingest-dbt -c /path/to/dbt-project
```

#### ⚙️ **Configuration:**
Users add OpenMetadata configuration directly to their `dbt_project.yml`:

```yaml
# dbt_project.yml

vars:
  # OpenMetadata configuration for metadata ingest-dbt command
  openmetadata_host_port: "https://your-openmetadata-url/your-endpoint" #(e.g.: http://localhost:8585/api)
  openmetadata_jwt_token: "your-jwt-token"
  openmetadata_service_name: "your-service-name"
```

#### 🧪 **Test Coverage:**
The comprehensive test suite validates:
- dbt_project.yml parsing and vars section validation
- OpenMetadata configuration extraction with all edge-case scenarios
- Variable format validation (naming conventions, URL formats)
- CLI execution and workflow configuration creation
  

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/tests/unit/test_dbt_ingest.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                'http://test-server:port/endpoint'
+            )
+            self.assertEqual(
+                workflow_config['workflowConfig']['openMetadataServerConfig']['securityConfig']['jwtToken'],
+                'test-jwt-token'
+            )
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (ulixius9) [inline ingestion/tests/unit/test_dbt_ingest.py]: this should not be required

File: ingestion/src/metadata/cli/ingest_dbt.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                    'searchAcrossDatabases': False,
+                    'databaseFilterPattern': {
+                        'includes': ['.*']
+                    },
+                    'schemaFilterPattern': {
+                        'includes': ['.*']
+                    },
+                    'tableFilterPattern': {
+                        'includes': ['.*']
+                    }
```
Comment (ulixius9) [inline ingestion/src/metadata/cli/ingest_dbt.py]: we should allow users to configure these values as well, we can make it optional and put default values if not provided

File: ingestion/src/metadata/cli/ingest_dbt.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            
+        raise ValueError(
+            f"Required OpenMetadata configuration not found in dbt_project.yml vars.\n"
+            f"Missing variables: {', '.join(missing_vars)}\n"
+            f"Please add the following to your dbt_project.yml:\n"
+            f"vars:\n"
+            f"  openmetadata_jwt_token: 'your-jwt-token'\n"
+            f"  openmetadata_host_port: 'your-host-port (e.g. http://openmetadata-server:8585/api)'\n"
+            f"  openmetadata_service_name: 'your-service-name'"
+        )
```
Comment (ulixius9) [inline ingestion/src/metadata/cli/ingest_dbt.py]: instead of doing this manual validation lets create a pedantic model and just parse the dictionary into the object

File: ingestion/src/metadata/cli/ingest_dbt.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    
+    return {
+        'jwt_token': jwt_token,
+        'host_port': host_port,
+        'service_name': service_name
+    }
+
```
Comment (ulixius9) [inline ingestion/src/metadata/cli/ingest_dbt.py]: and maybe let's directly return that object here for ease of access

**Review by ulixius9 — DISMISSED** [review]

**Comment by ulixius9** [comment]
@Khairajani the tests created by you are still failing can you resolve it?
